site/src/modules/workspaces/WorkspaceTiming/WorkspaceTimings.tsx

```diff
@@ -0,0 +1,214 @@
+import type { Interpolation, Theme } from "@emotion/react";
+import KeyboardArrowDown from "@mui/icons-material/KeyboardArrowDown";
+import KeyboardArrowUp from "@mui/icons-material/KeyboardArrowUp";
+import Button from "@mui/material/Button";
+import Collapse from "@mui/material/Collapse";
+import Skeleton from "@mui/material/Skeleton";
+import type { AgentScriptTiming, ProvisionerTiming } from "api/typesGenerated";
+import { type FC, useState } from "react";
+import { type TimeRange, calcDuration, mergeTimeRanges } from "./Chart/utils";
+import { ResourcesChart, isCoderResource } from "./ResourcesChart";
+import { ScriptsChart } from "./ScriptsChart";
+import { type StageCategory, StagesChart, stages } from "./StagesChart";
+
+type TimingView =
+	| { name: "default" }
+	| {
+			name: "detailed";
+			stage: string;
+			category: StageCategory;
+			filter: string;
+	  };
+
+type WorkspaceTimingsProps = {
+	defaultIsOpen?: boolean;
+	provisionerTimings: readonly ProvisionerTiming[] | undefined;
+	agentScriptTimings: readonly AgentScriptTiming[] | undefined;
+};
+
+export const WorkspaceTimings: FC<WorkspaceTimingsProps> = ({
+	provisionerTimings = [],
+	agentScriptTimings = [],
+	defaultIsOpen = false,
+}) => {
+	const [view, setView] = useState<TimingView>({ name: "default" });
+	const timings = [...provisionerTimings, ...agentScriptTimings];
+	const [isOpen, setIsOpen] = useState(defaultIsOpen);
+	const isLoading = timings.length === 0;
+
+	const displayProvisioningTime = () => {
+		const totalRange = mergeTimeRanges(timings.map(extractRange));
+		const totalDuration = calcDuration(totalRange);
+		return humanizeDuration(totalDuration);
+	};
+
+	return (
+		<div css={styles.collapse}>
+			<Button
+				disabled={isLoading}
+				variant="text"
+				css={styles.collapseTrigger}
+				onClick={() => setIsOpen((o) => !o)}
+			>
+				{isOpen ? (
+					<KeyboardArrowUp css={{ fontSize: 16, marginRight: 16 }} />
+				) : (
+					<KeyboardArrowDown css={{ fontSize: 16, marginRight: 16 }} />
+				)}
+				<span>Provisioning time</span>
```

Comment:
Nit: this is not just provisioning time since it also includes agent timings.
We should use a more descriptive name, like "Build timeline".